File: Easy/21. Merge Two Sorted Lists/ans.py
def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
	head = ListNode()
	curr = head
	while l1 and l2:
		while l1 and l2 and l1.val <= l2.val:
			curr.next = l1
			l1 = l1.next
			curr = curr.next
		while l1 and l2 and l2.val <= l1.val:
			curr.next = l2
			l2 = l2.next
			curr = curr.next
	curr.next = l1 or l2
	return head.next

File: Easy/83. Remove Duplicates from Sorted List/ans.js
var deleteDuplicates = function(head) {
    // the head points to the linked list
    if (head == null) return head
    
    // have a pointer to the head, which in this case is the full linked list
    let currHead = head
    
    while (currHead.next != null) {
        if (currHead.val == currHead.next.val) {
            // currHead doesn't change but the link/currHead.next would change and skip, so currHead could be seen as temporary unless solidfied in the else statement
            currHead.next = currHead.next.next
        } else {
            // if not equal to each other then currHead now links to currHead.next
            // solidfying the link
            currHead = currHead.next
        }
    }
    
    // after all the changes to the links made by the currHead, the head, which points to the entire linked list, outputs the new link list if there are changes made.
    return head    
};
